src/core/net/dnssd_server.cpp

```diff
@@ -856,6 +951,73 @@ Error Server::Response::ResolveBySrp(void)
     return error;
 }
 
+Error Server::Response::ResolveSoaOrNsQuery(void)
+{
+    Error error = kErrorNone;
+
+    // Handle AAAA (or ANY) query for the server host name used for
+    // SOA/NS answer.
+
+    if (mQuestions.IsFor(kRrTypeAaaa) || mQuestions.IsFor(kRrTypeAny))
+    {
+        Name::Buffer serverFullName;
+
+        ConstructFullName(Get<Server>().mSoaServerName, serverFullName);
+
+        if (QueryNameMatches(serverFullName))
+        {
+            SuccessOrExit(error = AppendAaaaRecord(Get<Mle::Mle>().GetMeshLocalEid(), kServerAaaaTtl));
+
+            for (Ip6::Netif::UnicastAddress &unicastAddr : Get<ThreadNetif>().GetUnicastAddresses())
+            {
+                if (!unicastAddr.mValid || !unicastAddr.mPreferred || unicastAddr.GetAddress().IsLinkLocalUnicast())
```

Comment:
The RLOC should also be skipped here, right? It's only for Thread internal (management) purposes. Maybe that's already the case here.

Reply:
I originally included all addresses (considering, why not? Since Thread nodes can use any address within the mesh, including ALOCs and RLOCs).

I've now changed it to behave like SRP (add non-mesh-local and non-link-local addresses); if none are available, add the ML-EID. Whatever list isis okay with me, as I don't think it particularly matters.